Odata-docs/performance/disable-writer-validations.md

```diff
@@ -0,0 +1,130 @@
+
+Use the [`ODataMessageWriterSettings.Validations`](/dotnet/api/microsoft.odata.odatamessagewritersettings.validations) property to configure which validations you want to enable or disable.
+
+You can disable all validations using [`ValidationKinds.None`]:
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.None;
+```
+
+You can selectively enable validations using the `&` operator:
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.ThrowOnDuplicatePropertyNames & ValidationKinds.ThrowIfTypeConflictsWithMetadata
+```
+
+You can enable all validations using [`ValidationKinds.All`]:
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.All;
+```
+
+To learn more about different validation options, consult the [`ValidationKinds` enum documentation](/dotnet/api/microsoft.odata.validationkinds).
+
+## Disabling writer validations when using `Microsoft.OData.Core` directly
+
+When using the core library directly, you pass the `ODataMessageWriterSettings` directly to the [`ODataMessageWriter`](/dotnet/api/microsoft.odata.odatamessagewriter) construtor.
+
+```csharp
+ODataMessageWriterSettings settings = new ODataMessageWriterSettings
+{
+    ODataUri = new ODataUri() { /* ... */ },
+    Validations = ValidationKinds.None
+};
+
+ODataMessageWriter writer = new ODataMessageWriter(odataMessage, settings, edmModel);
+
+// ...
+```
+
+## Disabling writer validations when using `Microsoft.AspNetCore.OData` 8
+
+In `Microsoft.AspNetCore.OData` 8, you can pass a service configuration action as the third argument of `AddRouteComponents`. Use
+this to inject a custom request-scoped `ODataMessageWriterSettings`, with validations disabled, to the service container:
+
+```csharp
+services
+.AddControllers()
+.AddOData(options =>
+    options.AddRouteComponents("routePrefix", edmModel, container =>
+    {
+        container.AddScoped<ODataMessageWriterSettings>(_ => new ODataMessageWriterSettings
+        {
+            Validations = ValidationKinds.None
+        })
+    }));
+```
+
+## Disabling writer validations when using `Microsoft.AspNetCore.OData` 7
+
+In `Microsoft.AspNetCore.OData` 7, you can pass a service configuration action to the `MapODataRoute` and `MapODataServiceRoute` methods.
+Use this to inject a custom request-scoped ODataMessageWriterSettings` to the service container.
+
+### Using endpoint routing
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+
+    // ...
+
+    app.UseEndpoints(endpoints =>
+    {
+        endpoints.MapODataRoute(
+            "odata", “odata”,
+            container =>
+            {
+                container.AddService<ODataMessageWriterSettings>(Microsoft.OData.ServiceLifetime.Scope, _ => new ODataMessageWriterSettings
+                {
+                    Validations = ValidationKinds.None
+                });
+
+                container.AddService(Microsoft.OData.ServiceLifetime.Singleton, sp => model);
```

Comment:
```suggestion
                container.AddService(Microsoft.OData.ServiceLifetime.Singleton, _ => model);
```
Because you used `_` 4 lines above....